forms.php

```diff
@@ -1,91 +1,160 @@
-            $mform->setType("course", PARAM_TEXT);
-
-            $options = ['optional' => true];
-            $mform->addElement("header", "section_timecompleted", get_string("form:section_timecompleted", "report_coursecompletion"));
-            $mform->setExpanded("section_timecompleted", false);
-            $mform->addElement('date_selector', 'timecompleted_after', get_string("form:timecompleted_after", "report_coursecompletion"), $options);
-            $mform->addElement('date_selector', 'timecompleted_before', get_string("form:timecompleted_before", "report_coursecompletion"), $options);
-
-            $mform->addElement("header", "section_timestarted", get_string("form:section_timestarted", "report_coursecompletion"));
-
-            $mform->addElement('date_selector', 'timestarted_after', get_string("form:timestarted_after", "report_coursecompletion"), $options);
-            $mform->addElement('date_selector', 'timestarted_before', get_string("form:timestarted_before", "report_coursecompletion"), $options);
-            $mform->closeHeaderBefore("search_operators");
-            $mform->setExpanded("section_timecompleted", false);
-
-            $radio_array = [];
-            $radio_array[] = $mform->createElement("radio", "operator", "", get_string("form:operator_and", "report_coursecompletion"), 0);
-            $radio_array[] = $mform->createElement("static", "space", "", "<br>");
-            $radio_array[] = $mform->createElement("radio", "operator", "", get_string("form:operator_or", "report_coursecompletion"), 1);
-            $mform->addGroup($radio_array, "search_operators", get_string("form:operator", "report_coursecompletion"), array(" "), false);
-            $mform->setDefault("operator", 0);
-
-            $this->add_action_buttons(false, get_string("form:search", "report_coursecompletion"));
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Coursecompletion report
+ *
+ * @package    report_coursecompletion
+ * @copyright  2017 Catalyst IT Ltd
+ * @author     Francis Devine <francis@catalyst.net.nz>
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+defined('MOODLE_INTERNAL') || die();
+
+require(__DIR__ . "/../../config.php");
```

Comment:
this line causes a fatal error on my machine. config.php should only be used in files loaded directly by the browser. Files should also either add the config or have the MOODLE_INTERNAL check - not both MOODLE_INTERNAL and require config.

for MOODLE_INTERNAL files, the require_login should also not be used - class files should only contain the class itself, and not hidden "in-line" code/api calls.

Also - this file "forms.php" is a class so should really be in the classes folder within the plugin and either within the plugin namespace, or use the plugin frankenstlye name as a prefix eg - the class name "ReportForm" is invalid - it should either be:
"report_coursecompletion_reportform"
or namespaced within report_coursecompletion: